src/ffishjs.cpp

```diff
@@ -263,6 +263,34 @@ class Board {
     return pos.game_ply();
   }
 
+  int game_result() const {
+    Value result;
+    bool gameEnd;
+    if (MoveList<LEGAL>(pos).size() > 0)
+      return VALUE_NONE;
+    gameEnd = pos.is_immediate_game_end(result);
+    if (!gameEnd)
+        result = pos.checkers() ? pos.checkmate_value() : pos.stalemate_value();
+    return result;
+  }
+
+  bool is_immediate_game_end() const {
+    return pos.is_immediate_game_end();
+  }
+
+  bool is_optional_game_end() const {
+    return pos.is_optional_game_end();
+  }
+
+  // Need to use different name than has_insufficient_material
+  bool has_insufficient_material_ffish(bool turn) const {
+    return has_insufficient_material(turn ? WHITE : BLACK, pos);
+  }
+
+  bool has_insufficient_material_ffish() const {
```

Comment:
Please remove suffix `_ffish`.
 
Below you can use
 ```C++
    return Stockfish::has_insufficient_material(WHITE, pos) && Stockfish::has_insufficient_material(BLACK, pos);
 ```
 to avoid ambiguity.